src/sentry/static/sentry/app/components/avatarCropper.tsx

```diff
@@ -0,0 +1,464 @@
+    });
+  };
+
+  // Normalize diff across dimensions so that negative diffs are always making
+  // the cropper smaller and positive ones are making the cropper larger
+  getDiffNW = (yDiff: number, xDiff: number) => {
+    return (yDiff - yDiff * 2 + (xDiff - xDiff * 2)) / 2;
+  };
+
+  getDiffNE = (yDiff: number, xDiff: number) => {
+    return (yDiff - yDiff * 2 + xDiff) / 2;
+  };
+
+  getDiffSW = (yDiff: number, xDiff: number) => {
+    return (yDiff + (xDiff - xDiff * 2)) / 2;
+  };
+
+  getDiffSE = (yDiff: number, xDiff: number) => {
+    return (yDiff + xDiff) / 2;
+  };
+
+  getNewDimensions = (container: HTMLDivElement, yDiff: number, xDiff: number) => {
+    const {resizeDimensions: oldDimensions, resizeDirection} = this.state;
+
+    const diff = this['getDiff' + resizeDirection!.toUpperCase()](yDiff, xDiff);
+
+    let height = container.clientHeight - oldDimensions.top;
+    let width = container.clientWidth - oldDimensions.left;
+
+    // Depending on the direction, we update different dimensions:
+    // nw: size, top, left
+    // ne: size, top
+    // sw: size, left
+    // se: size
+    const editingTop = resizeDirection === 'nw' || resizeDirection === 'ne';
+    const editingLeft = resizeDirection === 'nw' || resizeDirection === 'sw';
+
+    const newDimensions = {
+      top: 0,
+      left: 0,
+      size: oldDimensions.size + diff,
+    };
+
+    if (editingTop) {
+      newDimensions.top = oldDimensions.top - diff;
+      height = container.clientHeight - newDimensions.top;
+    }
+
+    if (editingLeft) {
+      newDimensions.left = oldDimensions.left - diff;
+      width = container.clientWidth - newDimensions.left;
+    }
+
+    if (newDimensions.top < 0) {
+      newDimensions.size = newDimensions.size + newDimensions.top;
+      if (editingLeft) {
+        newDimensions.left = newDimensions.left - newDimensions.top;
+      }
+      newDimensions.top = 0;
+    }
+
+    if (newDimensions.left < 0) {
+      newDimensions.size = newDimensions.size + newDimensions.left;
+      if (editingTop) {
+        newDimensions.top = newDimensions.top - newDimensions.left;
+      }
+      newDimensions.left = 0;
+    }
+
+    const maxSize = Math.min(width, height);
+    if (newDimensions.size > maxSize) {
+      if (editingTop) {
+        newDimensions.top = newDimensions.top + newDimensions.size - maxSize;
+      }
+      if (editingLeft) {
+        newDimensions.left = newDimensions.left + newDimensions.size - maxSize;
+      }
+      newDimensions.size = maxSize;
+    } else if (newDimensions.size < this.MIN_DIMENSION) {
+      if (editingTop) {
+        newDimensions.top = newDimensions.top + newDimensions.size - this.MIN_DIMENSION;
+      }
+      if (editingLeft) {
+        newDimensions.left = newDimensions.left + newDimensions.size - this.MIN_DIMENSION;
+      }
+      newDimensions.size = this.MIN_DIMENSION;
+    }
+
+    return {...oldDimensions, ...newDimensions};
+  };
+
+  validateImage() {
```

Comment:
@EvanPurkhiser I know in the original file this is how the function is named, but could we use this opportunity to change the name of this function to `validateImageOnUpload`? I remember when I played around with this file I was surprised that this function did not validate the image on cropping.